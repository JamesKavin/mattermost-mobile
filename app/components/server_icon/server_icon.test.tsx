// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {renderWithIntlAndTheme} from '@test/intl-test-helper';

import Icon from './index';

describe('Server Icon', () => {
    test('Server Icon Component should match snapshot', () => {
        const {toJSON} = renderWithIntlAndTheme(
            <Icon
                hasUnreads={false}
                mentionCount={0}
            />,
        );

        expect(toJSON()).toMatchSnapshot();
    });

    test('Server Icon Component should match snapshot with unreads', () => {
        const {toJSON} = renderWithIntlAndTheme(
            <Icon
                hasUnreads={true}
                mentionCount={0}
                testID='server_icon'
            />,
        );

        expect(toJSON()).toMatchSnapshot();
    });

    test('Server Icon Component should match snapshot with mentions', () => {
        const {toJSON} = renderWithIntlAndTheme(
            <Icon
                hasUnreads={false}
                mentionCount={1}
                testID='server_icon'
            />,
        );

        expect(toJSON()).toMatchSnapshot();
    });
});
