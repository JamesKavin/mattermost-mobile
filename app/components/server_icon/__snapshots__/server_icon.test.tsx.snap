// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server Icon Server Icon Component should match snapshot 1`] = `
<View>
  <View
    accessibilityState={
      {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    hitSlop={
      {
        "bottom": 5,
        "left": 40,
        "right": 20,
        "top": 20,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Icon
      color="rgba(255,255,255,0.56)"
      name="server-variant"
      size={24}
    />
  </View>
</View>
`;

exports[`Server Icon Server Icon Component should match snapshot with mentions 1`] = `
<View>
  <View
    accessibilityState={
      {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    hitSlop={
      {
        "bottom": 5,
        "left": 40,
        "right": 20,
        "top": 20,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="server_icon"
  >
    <Icon
      color="rgba(255,255,255,0.56)"
      name="server-variant"
      size={24}
    />
    <Text
      collapsable={false}
      numberOfLines={1}
      style={
        {
          "alignSelf": "flex-end",
          "backgroundColor": "#ffffff",
          "borderColor": "#14213e",
          "borderRadius": 11,
          "borderWidth": 2,
          "color": "#1e325c",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 12,
          "height": 22,
          "left": 13,
          "lineHeight": 16.5,
          "minWidth": 26,
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 5,
          "position": "absolute",
          "textAlign": "center",
          "top": -8,
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
      testID="server_icon.badge"
    >
      1
    </Text>
  </View>
</View>
`;

exports[`Server Icon Server Icon Component should match snapshot with unreads 1`] = `
<View>
  <View
    accessibilityState={
      {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    hitSlop={
      {
        "bottom": 5,
        "left": 40,
        "right": 20,
        "top": 20,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="server_icon"
  >
    <Icon
      color="rgba(255,255,255,0.56)"
      name="server-variant"
      size={24}
    />
    <Text
      collapsable={false}
      numberOfLines={1}
      style={
        {
          "alignSelf": "flex-end",
          "backgroundColor": "#ffffff",
          "borderColor": "#14213e",
          "borderRadius": 6,
          "borderWidth": 2,
          "color": "#1e325c",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 12,
          "height": 12,
          "left": 18,
          "lineHeight": 16.5,
          "minWidth": 12,
          "opacity": 1,
          "overflow": "hidden",
          "paddingHorizontal": 0,
          "position": "absolute",
          "textAlign": "center",
          "top": -5,
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
      testID="server_icon.badge"
    />
  </View>
</View>
`;
